### PR TITLE
[Refactor] Replace embedded Service with ServiceId

### PR DESCRIPTION
**Pull request by red-0ne**
## Summary

This PR replaces all embedded `Service` entries with `ServiceId`.

Most of the changes are `Service` -> `ServiceId` replacements but special attention has to be put on:
* x/session/types/session_header.go
* x/shared/helpers/service_configs.go
* x/shared/types/service.go
* x/service/types/message_add_service.go
* proto/poktroll/shared/service.proto
* proto/poktroll/session/types.proto
* proto/poktroll/session/query.proto

A follow-up to update `Shannon-SDK` is expected after approval.

E2E tests pass locally.

## Issue

image

## Type of change

Select one or more from the following:

- [ ] New feature, functionality or library
- [ ] Consensus breaking; add the `consensus-breaking` label if so. See #791 for details
- [ ] Bug fix
- [x] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

- [ ] **Documentation**: `make docusaurus_start`; only needed if you make doc changes
- [x] **Unit Tests**: `make go_develop_and_test`
- [x] **LocalNet E2E Tests**: `make test_e2e`
- [ ] **DevNet E2E Tests**: Add the `devnet-test-e2e` label to the PR.

## Sanity Checklist

- [x] I have tested my changes using the available tooling
- [ ] I have commented my code
- [x] I have performed a self-review of my own code; both comments & source code
- [ ] I create and reference any new tickets, if applicable
- [ ] I have left TODOs throughout the codebase, if applicable

### CODE REVIEW

File: x/shared/types/service.go [hunk 1]
```diff
@@ -3,9 +3,16 @@ package types
...
 	"net/url"
 	"regexp"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
 const (
+	// ComputeUnitsPerRelayMax is the maximum allowed compute_units_per_relay value when adding or updating a service.
+	// TODO_MAINNET: The reason we have a maximum is to account for potential integer overflows.
+	// Should we revisit all uint64 and convert them to BigInts?
```
Comment (Olshansk) [inline x/shared/types/service.go:13]: lol

**Review by Olshansk — APPROVED** [review]
That was much easier than expected, eh?

**Review by bryanchriswhite — COMMENTED** [review]
:ok_hand: Nice one @red-0ne! :sunglasses: 

It's interesting to see the effect this change has on the imports. I wonder if it has any effect on the module dependency tree. :thinking: It seems like we still need the `shared` module though as it currently contains the `Supplier` protobuf type as well.

File: x/proof/types/message_submit_proof_test.go [hunk 2]
```diff
@@ -70,17 +70,16 @@ func TestMsgSubmitProof_ValidateBasic(t *testing.T) {
...
 					SessionEndBlockHeight:   testsession.GetSessionEndHeightWithDefaultParams(1),
 				},
 				Proof: testClosestMerkleProof,
 			},
 			sessionHeaderToExpectedErrorFn: func(sh sessiontypes.SessionHeader) error {
-				serviceError := sharedtypes.ErrSharedInvalidService.Wrapf("invalid service ID: %s", sh.Service.Id)
-				sessionError := sessiontypes.ErrSessionInvalidService.Wrapf("invalid service: %v; %s", sh.Service, serviceError)
-				return ErrProofInvalidSessionHeader.Wrapf("invalid session header: %s", sessionError)
+				serviceError := sessiontypes.ErrSessionInvalidService.Wrapf("invalid service ID: %s", sh.ServiceId)
+				return ErrProofInvalidSessionHeader.Wrapf("invalid session header: %s", serviceError)
```
Comment (bryanchriswhite) [inline x/proof/types/message_submit_proof_test.go:82]:
```suggestion
				return ErrProofInvalidSessionHeader.Wrapf("%s", serviceError)
```

File: x/session/keeper/session_hydrator.go [hunk 3]
```diff
@@ -117,14 +117,14 @@ func (k Keeper) hydrateSessionID(ctx context.Context, sh *sessionHydrator) error
 	// TODO_MAINNET: In the future, we will need to validate that the Service is
 	// a valid service depending on whether or not its permissioned or permissionless
 
-	if err := sh.sessionHeader.Service.ValidateBasic(); err != nil {
-		return types.ErrSessionHydration.Wrapf("invalid service: %v; err %s", sh.sessionHeader.Service, err)
+	if !sharedtypes.IsValidServiceId(sh.sessionHeader.ServiceId) {
```
Comment (bryanchriswhite) [inline x/session/keeper/session_hydrator.go:120]: Nice reversal :joy::ok_hand: :dizzy:

File: x/session/types/errors.go [hunk 4]
```diff
@@ -16,4 +16,5 @@ var (
 	ErrSessionInvalidBlockHeight     = sdkerrors.Register(ModuleName, 1107, "invalid block height for session")
 	ErrSessionInvalidSessionId       = sdkerrors.Register(ModuleName, 1108, "invalid sessionId")
 	ErrSessionAppNotActive           = sdkerrors.Register(ModuleName, 1109, "application is not active")
+	ErrSessionServiceNotFound        = sdkerrors.Register(ModuleName, 1110, "service not found")
```
Comment (bryanchriswhite) [inline x/session/types/errors.go:19]: This doesn't seem to be used anywhere (according to my IDE). Is it mistaken?

File: x/shared/types/service.go [hunk 5]
```diff
@@ -3,9 +3,16 @@ package types
...
+	ComputeUnitsPerRelayMax uint64 = 2 ^ 16
```
Comment (bryanchriswhite) [inline x/shared/types/service.go:14]:
```suggestion
	ComputeUnitsPerRelayMax uint64 = 2 ^ 16  // 65536
```

**Review by bryanchriswhite — CHANGES_REQUESTED** [review]

**Review by bryanchriswhite — APPROVED** [review]
Preemptively approving. :+1: 

Just a few more suggestions regarding error messages; otherwise, LGTM. :ship:

**Comment by Olshansk** [comment]
Let's merge this in!

**Pull request merged**